library/core/src/ffi/c_str.rs

```diff
@@ -449,38 +445,59 @@ impl CStr {
     /// behavior when `ptr` is used inside the `unsafe` block:
     ///
     /// ```no_run
-    /// # #![allow(unused_must_use)]
-    /// # #![expect(dangling_pointers_from_temporaries)]
-    /// use std::ffi::CString;
-    ///
     /// // Do not do this:
-    /// let ptr = CString::new("Hello").expect("CString::new failed").as_ptr();
-    /// unsafe {
-    ///     // `ptr` is dangling
-    ///     *ptr;
-    /// }
+    /// let ptr = std::ffi::CStr::from_bytes_with_nul(
+    ///     &"Hello world!"
+    ///         .to_uppercase()
+    ///         .chars()
+    ///         .map(|c| c.try_into().unwrap())
```

Comment:
```suggestion
    ///         .bytes()
```

Reply:
I was worried this would break for non-ASCII chars...

Reply:
A CStr is a byte vector which does not contain `0x00` except for the terminator, it has no notion of "characters", despite whatever the C Standard might try to tell you.

Reply:
I definitely should have used `bytes` here, but it seems I was distracted by worrying about non-ASCII chars. Definitely a mistake on my part. Thanks for your keen attention.